src/Http/Http/src/StreamPipeWriter.cs

```diff
@@ -135,6 +136,12 @@ public override void Complete(Exception exception = null)
             }
 
             _currentSegmentOwner?.Dispose();
+
+            // We still want to cleanup segments before throwing an exception.
```

Comment:
Why? Why doesn't it just trigger the reader? That's what happens in the Pipe impl

Reply:
There is no reader, it would be a sync Write on the stream.

Reply:
Ah yes. Maybe this should just be IAsyncDuspoable

Reply:
The disposal timing would be problematic. Right now it's not disposed until OnCompleted. Because this is created dynamically nothing in pipeline knows to flush or dispose it before the response exits.

People must call FlushAsync after writing data or else that data will never be sent. At least this won't be flaky, it will always behave the same way and people should discover that during initial development.